cli/commands/deploy.ts

```diff
@@ -1,18 +1,63 @@
 import yargs, { Argv } from 'yargs'
 
-import { deployContract } from '../network'
+import {
+  getContractAt,
+  deployContract,
+  deployContractWithProxy,
+  deployContractAndSave,
+  deployContractWithProxyAndSave,
+} from '../network'
 import { loadEnv, CLIArgs, CLIEnvironment } from '../env'
 import { logger } from '../logging'
 
 export const deploy = async (cli: CLIEnvironment, cliArgs: CLIArgs): Promise<void> => {
   const contractName = cliArgs.contract
   const initArgs = cliArgs.init
+  const deployType = cliArgs.type
+  const buildAcceptProxyTx = cliArgs.buildTx
 
-  logger.log(`Deploying contract ${contractName}...`)
+  // Get the GraphProxyAdmin to own the GraphProxy for this contract
```

Comment:
I think this logic only needs to happen if deployType is deploy-proxy

Reply:
yes i see what you are saying